test.js

```diff
@@ -485,3 +493,64 @@ if (globalThis.AbortController !== undefined) {
 		});
 	});
 }
+
+async function collectAsyncIterable(asyncIterable) {
+	const values = [];
+
+	for await (const value of asyncIterable) {
+		values.push(value);
+	}
+
+	return values;
+}
+
+test('pMapIterable', async t => {
+	t.deepEqual(await collectAsyncIterable(pMapIterable(sharedInput, mapper)), [10, 20, 30]);
+});
+
+test('pMapIterable - empty', async t => {
+	t.deepEqual(await collectAsyncIterable(pMapIterable([], mapper)), []);
+});
+
+test('pMapIterable - iterable that throws', async t => {
+	let isFirstNextCall = true;
+
+	const iterable = {
+		[Symbol.asyncIterator]() {
+			return {
+				async next() {
+					if (!isFirstNextCall) {
+						return {done: true};
+					}
+
+					isFirstNextCall = false;
+					throw new Error('foo');
+				},
+			};
+		},
+	};
+
+	const iterator = pMapIterable(iterable, mapper)[Symbol.asyncIterator]();
+
+	await t.throwsAsync(iterator.next(), {message: 'foo'});
+});
+
+test('pMapIterable - mapper that throws', async t => {
+	await t.throwsAsync(collectAsyncIterable(pMapIterable(sharedInput, async () => {
+		throw new Error('foo');
+	})), {message: 'foo'});
+});
+
+test('pMapIterable - stop on error', async t => {
+	const output = [];
+
+	try {
+		for await (const value of pMapIterable(errorInput3, mapper)) {
+			output.push(value);
+		}
+	} catch (error) {
+		t.is(error.message, 'bar');
+	}
+
+	t.deepEqual(output, [20]);
+});
```

Comment:
There needs to be tests that proves it respects the `concurrency` and `backpressure` options.